taskcat/_cfn/template.py

```diff
@@ -151,8 +151,19 @@ def _find_children(self) -> None:  # noqa: C901
                 f"did not receive a valid template: {self.template_path} does not "
                 f"have a Resources section"
             )
-        for resource in self.template["Resources"].keys():
-            resource = self.template["Resources"][resource]
+        for resource_name, resource in self.template["Resources"].items():
+            if resource_name.startswith("Fn::ForEach::"):
+                for replicated_resource in resource[2].values():
```

Comment:
Index 2 of a "Fn:ForEach" is a map of ObjectKeys to ObjectValues (`Dict[str, Dict[str, dict]]). The [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-foreach.html#intrinsic-function-reference-foreach-declaration) are a bit ambiguous about the allowed number of items in the map, however [examples](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-foreach-example-resource.html#intrinsic-function-reference-foreach-example-replicate-multiple-resources) seem to imply multiple items are allowed in the map.

If needed, I can add a named constant here instead of hardcoding the index. Something like `FN_FOREACH_OUTPUT_MAP_INDEX`.